code/datums/components/lightdrawn.dm

```diff
@@ -0,0 +1,136 @@
+
+/**
+  * Drawn to light component
+  *
+  * This component adds game mechanics to emulate the moth chasing a lamp meme
+  */
+/datum/component/lightdrawn
+	/// Next time we can move
+	var/next_move
+	/// How often we move
+	var/move_delay = 2 SECONDS
+	/// Limiter to stop LAMP spam
+	var/next_message
+	/// current light mood
+	var/light_mood = 100
+	/// threshold below which lamp memes happen
+	var/light_mood_threshold = 50
+	/// threshold to compare with [/atom/movable/lighting_object/var/cached_max]
+	var/light_level_threshold = 1
+	/// boolean to track druggy/mesmerised overlay add/remove
+	var/mesmerised = FALSE
+
+/datum/component/lightdrawn/Initialize()
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	. = ..()
+
+	START_PROCESSING(SSprocessing, src)
+
+/datum/component/lightdrawn/Destroy(force, silent)
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/datum/component/lightdrawn/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_LIVING_FLASH_ACT, VARSET_CALLBACK(src, light_mood, 100))
+
+/datum/component/lightdrawn/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, COMSIG_LIVING_FLASH_ACT)
+
+/datum/component/lightdrawn/process()
+	var/turf/T = get_turf(parent)
+	if(!T)
+		return
+
+	var/mob/living/living_parent = parent
+
+	///The amount our ssight is tinted by, such as glasses
+	var/tint_factor = 1
+	///The brightness multplier for the seen brightness
+	var/brightness_mult = 1
+
+	if(iscarbon(parent))
+		var/mob/living/carbon/C = parent
+		var/totaltint = 0
+		if(C?.glasses)
+			totaltint += C.glasses.tint
+			brightness_mult = min(1, (2 - C.glasses.darkness_view) * 0.33)
+		var/obj/item/organ/eyes/eye = C.getorganslot(ORGAN_SLOT_EYES)
+		if(eye?.tint)
+			totaltint += eye.tint
+		tint_factor = 1/(1 + totaltint)

```

Comment:
`get_total_tint()` exists
Personally I'd register to `update_tint()` changes instead and update accordingly, saving this in a var to track it.